tests/test_datasets/test_dataset.py

```diff
@@ -619,6 +619,35 @@ def test_to_flatten_with_exog(df_and_regressors_flat):
     assert expected_df.equals(obtained_df)
 
 
+@pytest.mark.parametrize(
+    "columns, expected_columns",
+    (
+        (None, ["timestamp", "target", "segment", "regressor_1", "regressor_2"]),
+        (["regressor_2"], ["regressor_2"]),
+    ),
+)
+def test_to_flatten_correct_columns(df_and_regressors, columns, expected_columns):
+    df, df_exog, known_future = df_and_regressors
+    ts = TSDataset(df=df, df_exog=df_exog, freq="D", known_future=known_future)
+    flattened_df = ts.to_flatten(ts.df, columns=columns)
+    assert sorted(flattened_df.columns) == sorted(expected_columns)
```

Comment:
Looks like there should be `["timestamp", "segment", "regressor_2"]` at expected?

It seems we have kind of inconsistency between flatten and non flatten version - we have lost extra information but from the view of library -- `timestamp` and `segment` columns should be everywhere imho because we can't inverse to flatten without `timestamp` and `segment`

Reply:
I renamed column parameter to "features" and change it's typing to make it consistent with known_future